pkg/inmemorychannel/message_dispatcher.go

```diff
@@ -70,28 +69,25 @@ func (d *InMemoryDispatcher) Start(ctx context.Context) error {
 		break
 	}
 
-	// Done channel has been closed, we need to gracefully shutdown d.ceClient. The cancel() method will start its
+	// Done channel has been closed, we need to gracefully shutdown d.bindingsReceiver. The cancel() method will start its
 	// shutdown, if it hasn't finished in a reasonable amount of time, just return an error.
 	cancel()
 	select {
 	case err := <-errCh:
 		return err
 	case <-time.After(d.writeTimeout):
-		return errors.New("timeout shutting down ceClient")
+		return errors.New("timeout shutting http bindings receiver")
 	}
 }
 
-func NewDispatcher(args *InMemoryDispatcherArgs) *InMemoryDispatcher {
-	// TODO set read and write timeouts and port?
-	ceClient, err := kncloudevents.NewDefaultClient()
-	if err != nil {
-		args.Logger.Fatal("failed to create cloudevents client", zap.Error(err))
-	}
+func NewMessageDispatcher(args *InMemoryMessageDispatcherArgs) *InMemoryMessageDispatcher {
+	// TODO set read and write timeouts?
+	bindingsReceiver := kncloudevents.NewHttpMessageReceiver(args.Port)
```

Comment:
Could it be that we're not setting a sensible default for args.Port in our tests?

Reply:
Thanks for the idea. Seems port is set here https://github.com/knative/eventing/blob/master/pkg/reconciler/inmemorychannel/dispatcher/controller.go#L45, its set to 8080. I tried to change its value to 8081 and also tried to change the value in  https://github.com/knative/eventing/blob/master/pkg/channel/message_receiver.go#L70 but the tests fail nevertheless. If anyone has other ideas let me know!